bisection.py

```diff
@@ -237,22 +241,34 @@ def build(self, commit: Commit):
         version_py_path = os.path.join(self.srcpath, "torch/version.py")
         if os.path.exists(version_py_path):
             os.remove(version_py_path)
-        command = "python setup.py install"
-        subprocess.check_call(command, cwd=self.srcpath, env=build_env, shell=True)
+        try:
+            command = "python setup.py install"
+            subprocess.check_call(command, cwd=self.srcpath, env=build_env, shell=True)
+            command_testbuild = "python -c 'import torch'"
+            subprocess.check_call(command_testbuild, cwd=os.environ["HOME"], env=build_env, shell=True)
+        except subprocess.CalledProcessError:
+            # Remove the build directory, then try build it again
+            build_path = os.path.join(self.srcpath, "build")
+            if os.path.exists(build_path):
+                shutil.rmtree(build_path)
+            subprocess.check_call(command, cwd=self.srcpath, env=build_env, shell=True)
         print("done")
         # build pytorch lazy tensor if needed
         self._build_lazy_tensor(commit, build_env)
         self.build_install_deps(build_env)
 
-    def cleanup(self, commit: Commit):
-        print(f"Cleaning up packages from commit {commit.sha} ...", end="", flush=True)
+    def cleanup(self):
         packages = ["torch", "torchtext", "torchvision"]
-        command = "pip uninstall -y " + " ".join(packages) + " &> /dev/null "
-        subprocess.check_call(command, shell=True)
+        CLEANUP_ROUND = 5
+        # Clean up multiple times to make sure the packages are all uninstalled
+        for _ in range(CLEANUP_ROUND):
```

Comment:
Curious about this - there's a possibility of uninstall failing?

Reply:
To clarify, my question is why do we need 5 cleanup rounds?

Reply:
> To clarify, my question is why do we need 5 cleanup rounds?

This is because there might be multiple versions of staled torch packages in the environment - usually uninstalling twice is sufficient, but it does no harm to do it more times just to be safe.